src/openforms/authentication/utils.py

```diff
@@ -24,8 +25,19 @@ def store_auth_details(submission: Submission, form_auth: FormAuth) -> None:
         attribute in AuthAttribute.values
     ), f"Unexpected auth attribute {attribute} specified"
 
-    # DO NOT log this in plain text, it is considered sensitive information (BSN for example)
-    identifier = form_auth["value"]
-    submission.auth_plugin = plugin
-    setattr(submission, attribute, identifier)
-    submission.save(update_fields=["auth_plugin", attribute])
+    if hasattr(submission, "auth_info"):
+        # DO NOT log this in plain text, it is considered sensitive information (BSN for example)
+        auth_info = submission.auth_info
+        auth_info.plugin = plugin
+        auth_info.attribute = attribute
+        auth_info.value = form_auth["value"]
+        auth_info.save()
+        return
+
+    auth_data = AuthInfo(
+        plugin=plugin,
+        attribute=attribute,
+        value=form_auth["value"],
+        submission=submission,
+    )
+    auth_data.save()
```

Comment:
could be `AuthInfo.objects.update_or_create(submission=submission, defaults=...)` instead?